### PR TITLE
Remove Deprecated Usage Of `java.util.Date` API

### PR DESCRIPTION
**Pull request by isomarcte**

### CODE REVIEW

**Comment by isomarcte** [comment]
Build failure on appveyor is this.

```
X tests.CompletionSlowSuite.symbol-prefixes 4462ms 
  java.nio.file.FileSystemException: C:\projects\metals\tests\unit\target\e2e\completion\symbol-prefixes\.metals\metals.h2.db: The process cannot access the file because it is being used by another process.
  sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:86)
  sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:97)
  sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:102)
  sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:269)
  sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103)
  java.nio.file.Files.delete(Files.java:1126)
  scala.meta.internal.metals.RecursivelyDelete$$anon$1.visitFile(RecursivelyDelete.scala:26)
  scala.meta.internal.metals.RecursivelyDelete$$anon$1.visitFile(RecursivelyDelete.scala:20)
  java.nio.file.Files.walkFileTree(Files.java:2670)
  java.nio.file.Files.walkFileTree(Files.java:2742)
  scala.meta.internal.metals.RecursivelyDelete$.apply(RecursivelyDelete.scala:20)
  tests.BaseSlowSuite.cleanWorkspace(BaseSlowSuite.scala:94)
  tests.CompletionSlowSuite$.$anonfun$new$12(CompletionSlowSuite.scala:133)
  tests.BaseSlowSuite.tests$BaseSlowSuite$$loop$1(BaseSlowSuite.scala:103)
  tests.BaseSlowSuite$$anonfun$tests$BaseSlowSuite$$loop$1$1.applyOrElse(BaseSlowSuite.scala:106)
  tests.BaseSlowSuite$$anonfun$tests$BaseSlowSuite$$loop$1$1.applyOrElse(BaseSlowSuite.scala:103)
  scala.concurrent.Future.$anonfun$recoverWith$1(Future.scala:417)
  scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
  scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
  java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
  java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
  java.lang.Thread.run(Thread.java:748)
shutdown request received: build/shutdown
```

I don't think that is related to my changes.

File: metals-docs/src/test/scala/docs/SnapshotTest.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package docs
+
+import java.time._
+import utest._
+
+object SnapshotTest extends TestSuite {
```
Comment (olafurpg) [inline metals-docs/src/test/scala/docs/SnapshotTest.scala]: The `mtest` project in this build contains a `BaseSuite` that most of our test suites extends and exposes a `test(name: String): Unit` method. Most commonly, we structure test suites in this repo like here https://github.com/scalameta/metals/blob/6693914e4662220df4a04ce3ab841a11a851c0b0/tests/unit/src/test/scala/tests/TimerSuite.scala by creating a `def check(name: String, original: String, expected: ZonedDateTime): Unit` that is called like this
```scala
check(
  "basic",
  "Mon Apr 08 04:09:49 UTC 2019",
  ZonedDateTime.of(2019, 4, 8, 4, 9, 49, 0, ZoneId.of("UTC"))
)
// ...
```

**Comment by olafurpg** [comment]
The Appveyor CI has been unusually flaky recently, you can ignore the `CompletionSlowSuite` failure.

**Comment by isomarcte** [comment]
@olafurpg Thanks for the feedback. I updated the PR. I didn't see a meaningful way for the test name to vary, so I hard coded the `String`, but perhaps I misunderstood your intent. Let me know what you think and if you'd like me to change it.

**Comment by olafurpg** [comment]
FYI, I removed the added tests from the PR since I don't believe they are necessary and we don't run `docs/test` in the CI https://github.com/scalameta/metals/commit/333ab6fc00fb3542bcabd0dac51b91b72798768a

**Comment by isomarcte** [comment]
@olafurpg That's fine with me. I mostly just wanted to be sure I didn't modify the current formatting.

**Pull request merged**